pkg/awsutils/awsutils.go

```diff
@@ -920,6 +936,70 @@ func (cache *EC2InstanceMetadataCache) DeallocIPAddresses(eniID string, ips []st
 	return nil
 }
 
+func (cache *EC2InstanceMetadataCache) cleanUpLeakedENIs() {
+	rand.Seed(time.Now().UnixNano())
+	startupDelay := rand.Intn(eniCleanupStartupDelayMax)
+	log.Infof("Will attempt to clean up AWS CNI leaked ENIs after waiting %d seconds.", startupDelay)
+	time.Sleep(time.Duration(startupDelay) * time.Second)
+	// Loop forever
+	for {
+		networkInterfaces, err := cache.getFilteredListOfNetworkInterfaces()
+		if err != nil {
+			log.Warnf("Unable to get leaked ENIs: %v", err)
+		} else {
+			// Clean up all the leaked ones we found
+			for _, networkInterface := range networkInterfaces {
+				// Verify the description starts with "aws-K8S-"
+				if !strings.HasPrefix(aws.StringValue(networkInterface.Description), eniDescriptionPrefix) {
```

Comment:
Can you move this filtering into the `getFilteredListOfNetworkInterfaces()` to keep the logic clean?

Reply:
Good idea, will do :)